handlers/patchItem_test.go

```diff
@@ -0,0 +1,216 @@
+)
+
+func TestPatchItemMatch(t *testing.T) {
+	tests := []struct {
+		name        string
+		path        string
+		method      string
+		expectedRes bool
+	}{
+		{
+			name:        "Returns true for a matching path",
+			path:        "/lists/b6cf642d/items/73bb82c4/",
+			method:      "PATCH",
+			expectedRes: true,
+		},
+		{
+			name:        "Returns true for a uppercase ID",
+			path:        "/lists/B6CF642D/items/73BB82C4/",
+			method:      "PATCH",
+			expectedRes: true,
+		},
+		{
+			name:        "Returns true without trailing slash",
+			path:        "/lists/b6cf642d/items/73bb82c4",
+			method:      "PATCH",
+			expectedRes: true,
+		},
+		{
+			name:        "Returns false for list path",
+			path:        "/lists/b6cf642d/",
+			method:      "PATCH",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false for list path without trailing slash",
+			path:        "/lists/b6cf642d",
+			method:      "PATCH",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false for lists path",
+			path:        "/lists/",
+			method:      "PATCH",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false for lists path without trailing slash",
+			path:        "/lists",
+			method:      "PATCH",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false for items path",
+			path:        "/lists/b6cf642d/items/",
+			method:      "PATCH",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false when path is empty",
+			path:        "",
+			method:      "PATCH",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false for a GET request",
+			path:        "/lists/b6cf642d/items/73bb82c4/",
+			method:      "GET",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false for a POST request",
+			path:        "/lists/b6cf642d/items/73bb82c4/",
+			method:      "POST",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false for a DELETE request",
+			path:        "/lists/b6cf642d/items/73bb82c4/",
+			method:      "DELETE",
+			expectedRes: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			dbMocked := &mockDB{}
+			dbMocked.Test(t)
+			defer dbMocked.AssertExpectations(t)
+
+			input := createAPIGatewayV2HTTPRequest(tt.path, tt.method, "")
+			d := patchItem{db: dbMocked}
+			gotRes := d.match(input)
+
+			assert.Equal(t, tt.expectedRes, gotRes)
+		})
+	}
+}
+
+func TestPatchItemHandle(t *testing.T) {
+	tests := []struct {
+		name               string
+		path               string
+		listID             string
+		itemID             string
+		newName            string
+		body               string
+		output             *data.Item
+		outputErr          error
+		expectedRes        interface{}
+		expectedStatusCode int
+		shouldCallDB       bool
+	}{
+		{
+			name:               "Returns 'Bad Request' when the path is empty",
+			path:               "",
+			listID:             "test-list-id",
+			output:             nil,
+			outputErr:          nil,
```

Comment:
If `shouldCallDB` is false, you don't need to specify these

If you wanted, you could have a struct defined above `tests` that was something like:
```
type mockUpdateItem struct {
    output data.Item
    err error
}
```
and then in `tests` you could have `mockUpdateItem *mockUpdateItem`
You could then use `tt.mockUpdateItem != nil` in place of `shouldCallDB`

Up to you though. Just different ways of doing same thing